server/listener.go

```diff
@@ -16,22 +16,114 @@ package server
 
 import (
 	"net"
+
+	"golang.org/x/sync/errgroup"
 )
 
+type connRes struct {
+	conn net.Conn
+	err  error
+}
+
 type Listener struct {
-	net.Listener
-	h *Handler
+	netListener  net.Listener
+	fileListener net.Listener
+	eg           *errgroup.Group
+	conns        chan connRes
+	shutdown     chan struct{}
 }
 
 // NewListener creates a new Listener.
-func NewListener(protocol, address string, handler *Handler) (*Listener, error) {
-	l, err := net.Listen(protocol, address)
+func NewListener(protocol, address string, socket string) (*Listener, error) {
+	netl, err := net.Listen(protocol, address)
 	if err != nil {
 		return nil, err
 	}
-	return &Listener{l, handler}, nil
+
+	var unixl net.Listener
```

Comment:
btw... I saw that we check for GOOS in the Dolt codebase, but I still think we need something here, if nothing else, because some people do use go-mysql-server without Dolt.

Reply:
Yep! just added it :)